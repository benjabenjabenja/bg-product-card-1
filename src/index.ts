/**
 * ACA SE EXPORTAN TODOS LOS COMPONENTES, HOOKS, STYLES, Y LO QUE SEA QUE SE QUIERA EXPORTAR
 */

// COMPONENTES
export * from './components'

