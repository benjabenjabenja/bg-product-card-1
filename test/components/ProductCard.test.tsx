import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { ProductButtons, ProductCard, ProductImage, ProductTitle } from '../../src';
import { fakeMockProduct2 } from './data/products';

const product = fakeMockProduct2;
const increaseByNumber = 1;

describe('ProductCard', () => {

    test('should render component', () => { 
        const { container } = render(<ProductCard product={product}>
            {
                () => <h1>Product Card</h1>
            }
        </ProductCard>);

        expect(container).toMatchSnapshot();
    })
    
    test('should render component with image, title and buttons', () => {
        const { container } = render(<ProductCard product={product}>
            { 
                () => (
                    <>
                        <ProductImage img={product.img} />
                        <ProductTitle title={product.title} />
                        <ProductButtons />
                    </>
                )
            }
        </ProductCard>);
        expect(container).toMatchSnapshot();
    })

    test('', () => {
        const { container } = render(<ProductCard product={product}>
            {
                ({ count, increaseBy }) => (
                    <>
                        <h1>Product Card</h1>
                        <span>{count}</span>
                        <button onClick={() => increaseBy(increaseByNumber)}>+</button>
                    </>
                )
            }
        </ProductCard>);

        expect(container).toMatchSnapshot();

        const button = container.querySelector('button')!;
        expect(button).toBeDefined();
        expect(button?.textContent).toBe('+');
        expect(container.querySelector('button')?.getAttribute('onclick')).toBeDefined();

        fireEvent.click(button);
        expect(container.querySelector('span')?.textContent).toBe(increaseByNumber.toString());
    })
})

