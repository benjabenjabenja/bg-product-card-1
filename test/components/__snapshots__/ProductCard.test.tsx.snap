// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard  1`] = `
<div>
  <div
    class="productCard undefined"
  >
    <h1>
      Product Card
    </h1>
    <span>
      0
    </span>
    <button>
      +
    </button>
  </div>
</div>
`;

exports[`ProductCard should render component 1`] = `
<div>
  <div
    class="productCard undefined"
  >
    <h1>
      Product Card
    </h1>
  </div>
</div>
`;

exports[`ProductCard should render component with image, title and buttons 1`] = `
<div>
  <div
    class="productCard undefined"
  >
    <img
      alt="Product"
      class="productImg undefined"
      src="./cofee-mug.png"
    />
    <span
      class="productDescription undefined"
    >
      Product 2 - con imagen
    </span>
    <div
      class="buttonsContainer undefined"
    >
      <button
        class="buttonMinus"
      >
         - 
      </button>
      <div
        class="countLabel"
      >
         
        0
         
      </div>
      <button
        class="buttonAdd false "
      >
         + 
      </button>
    </div>
  </div>
</div>
`;
